src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -201,10 +201,14 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 
 		const parentDirPath = cwd.fsPath.split(resourceRequestConfig.pathSeparator).slice(0, -1).join(resourceRequestConfig.pathSeparator);
 		const parentCwd = URI.from({ scheme: cwd.scheme, path: parentDirPath });
+		const homeDir = cwd.with({ path: '' });
```

Comment:
hm, just realized this isn't quite right. we need to get the `$HOME` var, but it's not available on process here

Reply:
I guess we have to wait until `env` is provided via shell integration?

Reply:
that causes a layering issue, which is a bummer bc it works so nicely 😢

Reply:
@meganrogge I think `IShellEnvironmentService` or similar\* is what you want. To do this you'll need to create a base abstract TerminalCompletionService, then one for `browser` (that handles when HOME isn't available) and one for electron-sandbox (when HOME is available). Then pull via dependency injection and pass as args into `super()`. Look at `ITerminalProfileResolverService` for an example of how to do this.

\* I notice it's in electron-sandbox though so you'll also need to pull the environment in the case of remotes. Maybe hold off on that since we'll be moving on to the upcoming shell environment API stuff which will have it on a new capability.